mozmeao/birdbox
# Add support for custom social-sharing images

Added via the `wagtailmetadata` package (BSD licence), which gives us OG and sharing tags for FB, Twitter, etc via a simple templatetag and an additonal model field

We can't add the templatetag to the base template, because it will cause pages like 404.html to blwo up, because they don't have `self` in their context. So, instead, the tag has to be called on actual page templates, not a base.

Example of new image field in the UI

<img width="851" alt="Screenshot 2023-12-12 at 22 49 26" src="https://github.com/mozmeao/birdbox/assets/101457/b611b578-1b50-46b9-b2a7-c59fa9147e3b">



Example output:

<img width="1060" alt="Screenshot 2023-12-12 at 22 49 40" src="https://github.com/mozmeao/birdbox/assets/101457/7ccfd8de-88ff-4307-822e-7bf690c9e14b">


### To test locally:

* `just export-local-data` to back up your current local state
* check out branch
* ~`just migrate`~ `just preflight`
* Go to CMS admin and edit a page - on the Settings tab you'll see the new option to add a social image.
* Publish and view source to confirm you're getting a resized image as your sharing image
